### PR TITLE
TCM-800 |  Implement basic playback tracking

### PR DESCRIPTION
**Pull request by devender197**
### Summary

- [x] Tracked the video if 25% completed and 100% completed

### Issue

[TCM-800](https://hyfnla.atlassian.net/browse/TCM-800)

### Video/Screenshot
[Video Demo](https://drive.google.com/file/d/1r9u7r_nv0x043iwe5sVTdpBna5o21EYb/view?usp=sharing)

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt [hunk 1]
```diff
@@ -38,9 +41,15 @@ class VideoPlayerActivity : BrightcovePlayer() {
     private var videoSizeKnown = false
     private var videoWidth = 0
     private var videoHeight = 0
+    private var countDownTimer: CountDownTimer? = null
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt:44]: please rename if possible:
timer, is25PercentTracked, duration, videoCompleted

**Review by onodude — CHANGES_REQUESTED** [review]
thanks @devender197 added some comments around parity

File: app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt (hunk 1 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt]: [Done](https://github.com/hyfn/trx-consumer-android/pull/204/commits/90ef522ab9a04262ac053f8691885b4f98fba705)

File: app/src/main/java/com/trx/consumer/core/MainApplication.kt [hunk 2]
```diff
@@ -13,13 +14,24 @@ class MainApplication : Application() {
...
         if (BuildConfig.DEBUG) {
             Timber.plant(Timber.DebugTree())
         }
 
+        if (instance == null) instance = this
         configManager.configure(this)
     }
+
+    companion object {
+        private var instance: MainApplication? = null
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/core/MainApplication.kt:32]: Can do 
```
var instance: MainApplication? = null
   private set
   ```
to keep it more Kotlin

Comment (devender197) [reply app/src/main/java/com/trx/consumer/core/MainApplication.kt]: [Done](https://github.com/hyfn/trx-consumer-android/pull/204/commits/43fcf1a0805a8f33e35ad5bd3310d41a496cc774)

File: app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt [hunk 3]
```diff
@@ -198,10 +215,27 @@ class VideoPlayerActivity : BrightcovePlayer() {
         orientationJob?.apply { if (isActive) cancel() }
     }
 
+    private fun handleVideoProgress() {
+        timer = object : CountDownTimer(duration - videoCompleted, 10000) {
+            override fun onTick(millisUntilFinished: Long) {
+                videoCompleted = duration - millisUntilFinished
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt:221]: Dont think it's safe to use our own timer here, the video could stop playing for other reasons besides pressing pause that are out of our control like when it's buffering. Would be safer to use Brightcove's `EventType.PROGRESS` like in the first PR we had

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt:221]: https://github.com/hyfn/trx-consumer-android/blob/feature/%23504-player-analytics/app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Left a thought

File: app/src/main/java/com/trx/consumer/core/MainApplication.kt (hunk 2 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/core/MainApplication.kt]: Thank you!

File: app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt (hunk 3 above)
Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/screens/video/VideoPlayerActivity.kt:221]: @onodude Looks like iOS used the same timer approach, still have the same sentiment but your call

**Review by VivekV95 — APPROVED** [review]
Thanks for the changes!

**Review by onodude — APPROVED** [review]
thanks!

**Pull request merged**